datamol/viz/_lasso_highlight.py

```diff
@@ -0,0 +1,439 @@
+    if color_list is None:
+        color_list = [(0.5, 0.5, 0.5)] * len(indices_set_lists)
+    if len(color_list) < len(indices_set_lists):
+        raise ValueError("Not enough colors for referenced substructures!")
+
+    level_manager = defaultdict(set)
+    for match_atoms, color in zip(indices_set_lists, color_list):
+        used_levels = set.union(*[level_manager[a] for a in match_atoms])
+        if len(used_levels) == 0:
+            free_levels = {0}
+        else:
+            max_level = max(used_levels)
+            free_levels = set(range(max_level)) - used_levels
+
+        if free_levels:
+            draw_level = min(free_levels)
+        else:
+            draw_level = max(used_levels) + 1
+
+        for a in match_atoms:
+            level_manager[a].add(draw_level)
+
+        ar = r_min + r_dist * draw_level
+        _draw_substructurematch(
+            canvas,
+            mol,
+            match_atoms,
+            rel_radius=ar,
+            rel_width=max(relative_bond_width, ar),
+            color=color,
+            line_width=line_width,
+        )
+
+
+color_dict = {
+    "red": (1, 0, 0, 1),
+    "blue": (0, 0.5, 1, 1),
+    "orange": (1, 0.5, 0, 1),
+    "green": (0, 1, 0, 1),
+    "yellow": (1, 1, 0, 1),
+    "dark blue": (0, 0, 0.5, 1),
+}
+
+
+def lasso_highlight_image(
+    canvas_width: int,
+    canvas_height: int,
+    target_molecule: Union[str, dm.Mol],
+    search_molecules: Union[str, List[str], dm.Mol, List[dm.Mol]],
+) -> Image:
+    """A generalized interface to access both highlighting options whether the input is as a smiles, smarts or mol
+
+    Args:
+        canvas_width (int): _description_
+        canvas_height (int): _description_
+        target_molecule (Union[str, dm.Mol]):
+        search_molecules (Union[str, List[str], dm.Mol, List[dm.Mol]]): Various ways to enter but the search molecules must be entered as smart
+    """
+
+    # check if the input is valid
+    if target_molecule is None or (isinstance(target_molecule, str) and len(target_molecule) == 0):
+        raise ValueError("Please enter a valid target molecule or smiles")
+
+    if search_molecules is None or (
+        isinstance(search_molecules, str) and len(search_molecules) == 0
+    ):
+        raise ValueError("Please enter valid search molecules or smarts")
+
+    # less than 1 throws File parsing error: PNG header not recognized over 5,000 leads to a DecompressionBombError later on
+    if canvas_width < 1 or canvas_width > 5000 or canvas_height < 1 or canvas_height > 5000:
+        raise ValueError(
+            "To avoid errors please choose a number between 1-5000 for the canvas width or height"
+        )
+
+    if isinstance(target_molecule, str):
+        target_molecule = dm.to_mol(target_molecule)
+
+    mol = prepare_mol_for_drawing(target_molecule, kekulize=True)
+    d = rdMolDraw2D.MolDraw2DCairo(500, 500)
+    d.drawOptions().updateAtomPalette({i: (0, 0, 0, 1) for i in range(100)})
```

Comment:
It modifies the default rdkit atom colours. I would prefer sticking to the default. Eventually, you could add a boolean flag to enable it.

Reply:
leaving as default for now!